src/hooks/core/chat/useFetchChannelHook.ts

```diff
@@ -64,7 +64,7 @@ const useFetchChannelHook = () => {
         const cleanTopicName = newChannel?.id?.replace('topic_', '');
         const response = await getLatestTopicPost(cleanTopicName);
 
-        if (!response || response?.status !== 'success') return null;
+        if (channelType === 'topics' && (!response || response?.status !== 'success')) return null;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The change made here is logical and improves the code by adding a check for `channelType === 'topics'` before returning null. This ensures that the function only returns null when it's dealing with topics and the response status is not 'success'. However, it would be better to handle this error more explicitly rather than just returning null. Consider throwing an error or logging the issue for better debugging.
```diff
- if (channelType === 'topics' && (!response || response?.status !== 'success')) return null;
+ if (channelType === 'topics' && (!response || response?.status !== 'success')) {
+   console.error(`Failed to fetch channel of type ${channelType}`);
+   return null;
+ }
```
This way, you can track the occurrences of this issue in your logs and debug it more effectively if needed.


<!-- This is an auto-generated comment by OSS CodeRabbit -->